WalletWasabi.Fluent/ViewModels/AddWallet/AddWalletPageViewModel.cs

```diff
@@ -22,68 +23,65 @@ namespace WalletWasabi.Fluent.ViewModels.AddWallet;
 	NavBarPosition = NavBarPosition.Bottom)]
 public partial class AddWalletPageViewModel : DialogViewModelBase<Unit>
 {
-	[AutoNotify] private string _walletName = "";
-
 	public AddWalletPageViewModel()
 	{
-		SelectionMode = NavBarItemSelectionMode.Button;
+		CreateWalletCommand = ReactiveCommand.Create(OnCreateWallet);
 
-		var canExecute = this.WhenAnyValue(x => x.WalletName)
-			.ObserveOn(RxApp.MainThreadScheduler)
-			.Select(x => !string.IsNullOrWhiteSpace(x) && !Validations.Any);
+		ConnectHardwareWalletCommand = ReactiveCommand.Create(OnConnectHardwareWallet);
 
-		NextCommand = ReactiveCommand.Create(OnNext, canExecute);
+		ImportWalletCommand = ReactiveCommand.CreateFromTask(async () => await OnImportWalletAsync());
 
-		OpenCommand = ReactiveCommand.Create(
-			async () =>
-		{
-			MainViewModel.Instance.IsOobeBackgroundVisible = true;
+		RecoverWalletCommand = ReactiveCommand.Create(OnRecoverWallet);
 
-			await NavigateDialogAsync(this, NavigationTarget.DialogScreen);
+		OpenCommand = ReactiveCommand.Create(async () => await NavigateDialogAsync(this, NavigationTarget.DialogScreen));
+	}
 
-			MainViewModel.Instance.IsOobeBackgroundVisible = false;
-		});
+	public ICommand CreateWalletCommand { get; }
 
-		this.ValidateProperty(x => x.WalletName, errors => ValidateWalletName(errors, WalletName));
-	}
+	public ICommand ConnectHardwareWalletCommand { get; }
 
-	private void OnNext()
+	public ICommand ImportWalletCommand { get; }
+
+	public ICommand RecoverWalletCommand { get; }
+
+	private void OnCreateWallet()
 	{
-		Navigate().To(new SelectWalletCreationOptionViewModel(WalletName));
+		Navigate().To(new WalletNamePageViewModel(WalletCreationOption.AddNewWallet));
 	}
 
-	private static void ValidateWalletName(IValidationErrors errors, string walletName)
+	private void OnConnectHardwareWallet()
 	{
-		string walletFilePath = Path.Combine(Services.WalletManager.WalletDirectories.WalletsDir, $"{walletName}.json");
+		Navigate().To(new WalletNamePageViewModel(WalletCreationOption.ConnectToHardwareWallet));
+	}
 
-		if (string.IsNullOrEmpty(walletName))
+	private async Task OnImportWalletAsync()
+	{
+		try
 		{
-			return;
-		}
+			var filePath = await FileDialogHelper.ShowOpenFileDialogAsync("Import wallet file", new[] { "json" });
 
-		if (walletName.IsTrimmable())
-		{
-			errors.Add(ErrorSeverity.Error, "Leading and trailing white spaces are not allowed!");
-			return;
-		}
+			if (filePath is null)
+			{
+				return;
+			}
 
-		if (File.Exists(walletFilePath))
-		{
-			errors.Add(
-				ErrorSeverity.Error,
-				$"A wallet named {walletName} already exists. Please try a different name.");
-			return;
-		}
+			var walletName = Path.GetFileNameWithoutExtension(filePath);
```

Comment:
@soosr should be fixed now. 

Import wallet will run the same validations as create new, try to infer wallet name from filename (otherwise ask the user) and gracefully catch any import errors